sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -267,3 +301,93 @@ def get_region_zones_for_accelerators(
+            # NOTE: While it is allowed to use A2 machines as CPU-only nodes,
+            # we exclude this case as it is uncommon and undesirable.
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.ResourcesMismatchError(
+                    'A2 instance types should be used with A100 GPUs. '
+                    'Either use other instance types or specify the '
+                    'accelerators as A100.')
+        return
+
+    acc = list(accelerators.items())
+    assert len(acc) == 1, acc
+    acc_name, acc_count = acc[0]
+
+    if acc_name.startswith('tpu-'):
+        if instance_type != 'TPU-VM' and not instance_type.startswith('n1-'):
+            # TODO(woosuk): Check max vcpus and memory for each TPU type.
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.ResourcesMismatchError(
+                    'TPU Nodes can be only used with N1 machines. '
+                    'Please refer to: '
+                    'https://cloud.google.com/compute/docs/general-purpose-machines#n1_machines')  # pylint: disable=line-too-long
+        return
+
+    # Treat A100 as a special case.
+    if acc_name == 'A100':
+        # A100 must be attached to A2 instance type.
+        a100_instance_type = _A100_INSTANCE_TYPES[acc_count]
+        if not instance_type.startswith('a2-'):
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.ResourcesMismatchError(
+                    f'A100 GPUs cannot be attached to {instance_type}. '
+                    f'Use {a100_instance_type} instead. Please refer to '
+                    'https://cloud.google.com/compute/docs/gpus#a100-gpus')
+        if instance_type != a100_instance_type:
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.ResourcesMismatchError(
+                    f'A100:{acc_count} cannot be attached to {instance_type}. '
+                    f'Use {a100_instance_type} instead. Please refer to '
+                    'https://cloud.google.com/compute/docs/gpus#a100-gpus')
+        return
+
+    # Other GPUs must be attached to N1 machines.
+    # Refer to: https://cloud.google.com/compute/docs/machine-types#gpus
+    if not instance_type.startswith('n1-'):
+        with ux_utils.print_exception_no_traceback():
+            raise exceptions.ResourcesMismatchError(
+                f'{acc_name} GPUs cannot be attached to {instance_type}. '
+                'Use N1 instance types instead. Please refer to: '
+                'https://cloud.google.com/compute/docs/machine-types#gpus')
+
+    # Check maximum vCPUs and memory.
+    if acc_name not in _NUM_ACC_TO_MAX_CPU_AND_MEMORY:
+        with ux_utils.print_exception_no_traceback():
+            raise exceptions.ResourcesUnavailableError(
+                f'{acc_name} is not available in GCP. '
+                'See \'sky show-gpus --cloud gcp\'')
```

Comment:
Should this be `assert acc_name in _NUM_ACC_TO_MAX_CPU_AND_MEMORY`?

It should've been caught outside. E.g., under this branch
```
» sky launch --cloud gcp --gpus M60  ''                                                         1 ↵
I 08-29 15:02:52 optimizer.py:879] No resource satisfying {'M60': 1} on [GCP].
sky.exceptions.ResourcesUnavailableError: No launchable resource found for task sky-cmd. To fix: relax its resource requirements.
Hint: 'sky show-gpus --all' to list available accelerators.
      'sky check' to check the enabled clouds.
```

Reply:
That's a good point. Actually, `--cloud gcp --gpus M60 ''` and ` --instance-type n1-highmem-8 --gpus M60` will raise different error messages:

1.
```
$ sky launch --cloud gcp --gpus M60 ''
I 08-30 22:06:54 optimizer.py:875] No resource satisfying {'M60': 1} on [GCP].
sky.exceptions.ResourcesUnavailableError: No launchable resource found for task sky-cmd. To fix: relax its resource requirements.
Hint: 'sky show-gpus --all' to list available accelerators.
      'sky check' to check the enabled clouds.
```

2.
```
$ sky launch --instance-type n1-highmem-8 --gpus M60 ''
sky.exceptions.ResourcesUnavailableError: M60 is not available in GCP. See 'sky show-gpus --cloud gcp'
```

In the first case, the optimizer asks itself which instance to choose, and finds that GCP does not support M60. On the other hand, in the second case, the optimizer checks whether M60 can be attached to n1-highmem-8, and the new checks added in `gcp_catalog` finds the error. Since the two cases take different paths, the error messages are different.